src/topology/sheaves/presheaf_of_functions.lean

```diff
@@ -34,6 +34,27 @@ namespace Top
 
 variables (X : Top.{v})
 
+/--
+The presheaf of dependently typed functions on `X`, with fibres given by a type family `f`.
+There is no requirement that the functions are continuous, here.
+-/
+def presheaf_to_Types (f : X → Type v) : X.presheaf (Type v) :=
+{ obj := λ U, Π x : (unop U), f x,
+  map := λ U V i g, λ (x : unop V), g (i.unop x) }
+
+/--
+The presheaf of functions on `X` with values in a type `T`.
+There is no requirement that the functions are continuous, here.
+-/
+def presheaf_to_Type (T : Type v) : X.presheaf (Type v) :=
```

Comment:
Does this need a separate definition? Can it be defined in terms of `presheaf_to_Types`?

Reply:
Unfortunately it does need a separate definition: it matter latter that Lean can see syntactically that on objects we get non-dependent functions.

I've added a comment explaining this.